### PR TITLE
Consolidate Server

### PR DESCRIPTION
**Pull request by TheTedder**
This PR would merge all LiveSplit.Server functionality into the internal server. As a security measure, it also removes remote script execution capabilities that the server previously had.

### CODE REVIEW

File: LiveSplit/LiveSplit.Tests/TimeFormatTests/DeltaTimeFormattersTests.cs [hunk 1]
```diff
@@ -133,41 +133,5 @@ public void DeltaSplitTimeFormatterFormatsTimeCorrectly_WhenTimeIsValid(string t
             var formattedTime = sut.Format(time);
             Assert.Equal(expectedDelta, formattedTime);
         }
-
-        [Fact]
-        public void PreciseDeltaFormatterFormatTimeAsDash_WhenTimeIsNull()
-        {
```
Comment (wooferzfg) [inline LiveSplit/LiveSplit.Tests/TimeFormatTests/DeltaTimeFormattersTests.cs:139]: Why remove these?

Comment (TheTedder) [reply LiveSplit/LiveSplit.Tests/TimeFormatTests/DeltaTimeFormattersTests.cs:139]: The class they test was part of Livesplit.Server and was made redundant when I added the new time formatter.

**Comment by wooferzfg** [comment]
Since we're deprecating LiveSplit.Server with this PR, could you:
- Update the README of LiveSplit.Server to indicate that it's deprecated now
- Add the server documentation to this repo

File: LiveSplit/LiveSplit.View/View/TimerForm.cs [hunk 2]
```diff
@@ -316,8 +317,7 @@ private void Init(string splitsPath = null, string layoutPath = null)
             BackColor = Color.Black;
 
             Server = new CommandServer(CurrentState);
-            Server.Start();
-
+            Server.StartNamedPipe();
```
Comment (wooferzfg) [inline LiveSplit/LiveSplit.View/View/TimerForm.cs:320]: Keep the extra newline below this line please

File: LiveSplit/LiveSplit.View/View/TimerForm.cs [hunk 3]
```diff
@@ -681,6 +681,28 @@ void CurrentState_OnUndoSplit(object sender, EventArgs e)
...
         }
 
+        void ServerMenuItem_Click(object sender, EventArgs e)
+        {
+            if (ServerStarted)
+            {
+                Server.StopTcp();
+                this.InvokeIfRequired(() =>
+                {
+                    serverMenuItem.Text = "Start Server";
```
Comment (wooferzfg) [inline LiveSplit/LiveSplit.View/View/TimerForm.cs:691]: Should we change this to "Start TCP Server" to differentiate between the old server component text and the new text, and also add some clarity?

**Comment by wooferzfg** [comment]
Please update this line of the README https://github.com/LiveSplit/LiveSplit/blob/277cc24e43d4a50bf55cca08d779cc9110e8c786/README.md#L74

**Comment by wooferzfg** [comment]
If I open the Settings dialog, click "Cancel", and then open it again, the Server Port gets set to 0.

File: LiveSplit/LiveSplit.View/View/TimerForm.cs (hunk 3 above)
Comment (TheTedder) [reply LiveSplit/LiveSplit.View/View/TimerForm.cs:691]: Sure, sounds good.

File: README.md [hunk 4]
```diff
@@ -65,13 +65,119 @@ The documentation for how to develop, test, and submit an Auto Splitter can be f
 
 [Auto Splitters Documentation](https://github.com/LiveSplit/LiveSplit.AutoSplitters/blob/master/README.md)
 
+## The LiveSplit Server
+
+The internal LiveSplit Server allows for other programs and other computers to control LiveSplit. The server can accept connections over either a named pipe located at `\\<hostname>\pipe\LiveSplit` (`.` is the hostname if the client and server are on the same computer) or over TCP/IP.
+
+### Control
+
+The named pipe is always open while LiveSplit is running but the TCP server **MUST** be started before programs can talk to it (Right click on LiveSplit -> Control -> Start Server). You **MUST** manually start it each time you launch LiveSplit.
```
Comment (wooferzfg) [inline README.md:74]:
```suggestion
The named pipe is always open while LiveSplit is running but the TCP server **MUST** be started before programs can talk to it (Right click on LiveSplit -> Control -> Start TCP Server). You **MUST** manually start it each time you launch LiveSplit.
```

File: README.md [hunk 5]
```diff
@@ -65,13 +65,119 @@ The documentation for how to develop, test, and submit an Auto Splitter can be f
...
+
+### Settings
+
+#### Port
+
+**Port** is the door (one of thousands) on your computer that this program sends data through. Default is 16834. This should be fine for most people, but depending on network configurations, some ports may be blocked. See also https://en.wikipedia.org/wiki/Port_%28computer_networking%29
```
Comment (wooferzfg) [inline README.md:80]:
```suggestion
**Server Port** is the door (one of thousands) on your computer that this program sends data through. Default is 16834. This should be fine for most people, but depending on network configurations, some ports may be blocked. See also https://en.wikipedia.org/wiki/Port_%28computer_networking%29
```

File: README.md [hunk 6]
```diff
@@ -65,13 +65,119 @@ The documentation for how to develop, test, and submit an Auto Splitter can be f
...
+
+The internal LiveSplit Server allows for other programs and other computers to control LiveSplit. The server can accept connections over either a named pipe located at `\\<hostname>\pipe\LiveSplit` (`.` is the hostname if the client and server are on the same computer) or over TCP/IP.
+
+### Control
+
+The named pipe is always open while LiveSplit is running but the TCP server **MUST** be started before programs can talk to it (Right click on LiveSplit -> Control -> Start Server). You **MUST** manually start it each time you launch LiveSplit.
+
+### Settings
+
+#### Port
```
Comment (wooferzfg) [inline README.md:78]:
```suggestion
#### Server Port
```

**Comment by wooferzfg** [comment]
Thanks!

**Pull request merged**